### DISCUSSION TITLE
Is there a way to directly write models in Java "by hand"?

### DISCUSSION THREAD

**Discussion by luxspes** [oracle/sandwood#6] [Q&A]
From what I read, one can create models using the Sandwook language with a non java syntax. Is there also a way to directly write models in Java "by hand"? (Just as one can write them in Scala when using https://github.com/charles-river-analytics/figaro )

**Comment by Daniel-Goodman** [comment]
Currently there are 2 ways to write Sandwood models: 
1. In the Sandwood language and this is the way we would expect people to right models. 
2. By writing the Java API calls that are used to construct the DAG representation of the code. This could be used to write models by hand, but it is not recommended. The Sandwood parser translates Sandwood models into these API calls before the rest of the compiler is triggered. This API is available, and while human readable is not intended for people to write by hand, but is exposed to allow transformations from other languages.

A future goal is to work with [project Babylon](https://openjdk.org/projects/babylon/) to allow Sandwood models to be written directly from Java.

One important difference between Sandwood and Figaro is that Sandwood is compiled while Figero appears to just be constructed from API calls in a host language (Scala). This forces certain trade offs.
